File: frontend/ClickAndDrive/src/app/services/panic.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

export interface PanicRequest {
  rideId: number;
  guest: boolean;
  userId: number;
}

export interface PanicResponse {
  id: number;
  rideId: number;
  isGuest: boolean;
  triggeredByName: string;
  triggeredByEmail: string;
  createdAt: string;
  resolved: boolean;
  originAddress: string;
  destinationAddress: string;
}

@Injectable({ providedIn: 'root' })
export class PanicService {
  private apiUrl = 'http://localhost:8080/api/panic';

  constructor(private http: HttpClient) {}

  triggerPanic(request: PanicRequest): Observable<PanicResponse> {
    console.log('PanicService - sending request:', request);
    return this.http.post<PanicResponse>(this.apiUrl, request);
  }

  getAllPanics(): Observable<PanicResponse[]> {
    return this.http.get<PanicResponse[]>(`${this.apiUrl}/all`);
  }

  getUnresolvedPanics(): Observable<PanicResponse[]> {
    return this.http.get<PanicResponse[]>(`${this.apiUrl}/unresolved`);
  }

  resolvePanic(panicId: number): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${panicId}/resolve`, {});
  }
}

File: mobile-version/Navbar-shortened/app/src/main/java/com/example/clickanddrive/clients/services/PanicService.java
package com.example.clickanddrive.clients.services;

import com.example.clickanddrive.dtosample.requests.PanicRequest;
import com.example.clickanddrive.dtosample.responses.PanicResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface PanicService {

    @POST("panic")
    Call<PanicResponse> triggerPanic(@Body PanicRequest request);

    @GET("panic/unresolved")
    Call<List<PanicResponse>> getUnresolvedPanics();

    @GET("panic/all")
    Call<List<PanicResponse>> getAllPanics();

    @PUT("panic/{id}/resolve")
    Call<Void> resolvePanic(@Path("id") Long panicId);
}